language/testing-infra/e2e-tests/src/account.rs

```diff
@@ -85,6 +85,22 @@ impl Account {
         }
     }
 
+    /// Creates a new account with the given addr and key pair
+    ///
+    /// Like with [`Account::new`], the account returned by this constructor is a purely logical
+    /// entity.
+    pub fn new_validator(
+        addr: AccountAddress,
+        privkey: Ed25519PrivateKey,
+        pubkey: Ed25519PublicKey,
+    ) -> Self {
+        Account {
+            addr,
+            privkey,
+            pubkey,
+        }
+    }
```

Comment:
Don't we assume that the account is derived from the pubkey a couple places?  Because I know the `with_keypair()` does the same as this but derive the account.  Oh but I see you're trying to change it.